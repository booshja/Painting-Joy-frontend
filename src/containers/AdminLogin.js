// dependencies
import React from "react";
import styled from "styled-components";
// components
import { AdminPageTitle, LoadingSpinner } from "../components";
import { AdminHeader } from "../containers";
import { Redirect } from "react-router-dom";
// context
import { useAuth0 } from "@auth0/auth0-react";
// hooks
import { useHistory } from "react-router-dom";

const StyledAdminLogin = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;
const StyledP = styled.p`
    text-align: center;
    margin-top: 50vh;
    font-family: "Barlow Condensed", sans-serif;
    font-size: 1rem;
`;

const StyledLoginButton = styled.button`
    border: 2px solid #207070;
    border-radius: 4px;
    padding: 0.5px 8px;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1.4rem;
    letter-spacing: 1px;
    cursor: pointer;
    background-color: #207070;
    color: #ffffff;

    &:hover {
        background-color: #ffffff;
        color: #207070;
    }
`;

const AdminLogin = () => {
    // set up auth context
    const { loginWithRedirect, isLoading, isAuthenticated } = useAuth0();
    // set up history
    const history = useHistory();

    if (isLoading)
        return (
            <StyledAdminLogin>
                <LoadingSpinner />
            </StyledAdminLogin>
        );

    if (isAuthenticated) return <Redirect to="/admin/dashboard" />;

    return (
        <>
            <AdminHeader login={true} />
            <main>
                <StyledAdminLogin>
                    <AdminPageTitle>Login</AdminPageTitle>
                    <StyledLoginButton onClick={() => loginWithRedirect()}>
                        Click Here To Log In
                    </StyledLoginButton>
                </StyledAdminLogin>
            </main>
            <footer>
                <StyledP>
                    Copyright &copy; 2021 - Painting Joy Mural Co.
                </StyledP>
            </footer>
        </>
    );
};

export default AdminLogin;
