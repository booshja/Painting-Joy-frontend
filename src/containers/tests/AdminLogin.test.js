import { render } from "@testing-library/react";
import AdminLogin from "../AdminLogin";

/** Smoke Test */
it("renders without crashing", () => {
    render(<AdminLogin />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<AdminLogin />);
    expect(asFragment()).toMatchSnapshot();
});
