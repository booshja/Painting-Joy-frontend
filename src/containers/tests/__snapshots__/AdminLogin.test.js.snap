// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  color: #207070;
  font-size: 5rem;
  margin-top: 2rem;
}

.c2 {
  font-size: 1.4rem;
  color: #207070;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-top: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  min-height: 100vh;
  width: 100%;
}

@media screen and (min-width:768px) {
  .c0 {
    padding: 0 25%;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding: 0 31%;
  }
}

@media screen and (min-width:1280px) {
  .c0 {
    padding: 0 35%;
  }
}

@media screen and (min-width:1440px) {
  .c0 {
    padding: 0 6.6rem;
  }
}

@media screen and (min-width:1440px) {
  .c0 {
    width: 600px;
  }
}

<div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-palette fa-w-16 fa-spin c1"
      data-icon="palette"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M204.3 5C104.9 24.4 24.8 104.3 5.2 203.4c-37 187 131.7 326.4 258.8 306.7 41.2-6.4 61.4-54.6 42.5-91.7-23.1-45.4 9.9-98.4 60.9-98.4h79.7c35.8 0 64.8-29.6 64.9-65.3C511.5 97.1 368.1-26.9 204.3 5zM96 320c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm32-128c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128-64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
        fill="currentColor"
      />
    </svg>
    <p
      class="c2"
    >
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
